lib/matplotlib/layout_engine.py

```diff
@@ -101,6 +101,28 @@ def execute(self, fig):
         raise NotImplementedError
 
 
+class NullLayoutEngine(LayoutEngine):
```

Comment:
Do we need to have this public?

Can we think of a better name? Something that catches "mirror the behavior of whatever layout engine it is replacing"? Maybe FrozenLayoutEngine or LayoutEngineSnapshot? Also this behavior is the key reason for the existence of the class. It should be prominently in the docstring, not only in the parameter description.

Reply:
This turns the layout engine off, so I don't think it really freeze the layout manager?

Reply:
There is a difference between `None` and `NullLayoutEngine`, and we have to capture that at least in the docstring but if possible already in the class name. `NullLayoutEngine` sounds as if it does nothing at all and intuitively I'd have expected that that's equvalent with the initial state `None`.

Reply:
`NullLayoutEngineLockedColorbarImplimentation` is a bit long!

Reply:
BTW, I think this needs to be public because the user can get it as a result from `get_layout_engine`.  Maybe what is needed is a sentence of explanation for the user in the docs....